test/jit/test_freezing.py

```diff
@@ -2189,78 +2189,89 @@ def test_conv_to_mkldnn_no_mkldnn(self):
             inp = torch.rand([4, 3, 4, 4])
             self.assertEqual(frozen(inp), mod(inp))
 
-    @unittest.skipIf(not TEST_CUDNN, "requires CUDNN")
+    @unittest.skipIf(not (TEST_CUDNN or TEST_WITH_ROCM), "requires CUDNN")
     def test_freeze_conv_relu_fusion(self):
-        conv_bias = [True, False]
-        conv_ops = [nn.Conv2d, nn.Conv3d]
-        add_z = [True, False]
-        use_tracing = [True, False]
-        for use_bias, conv, add_z, tracing in product(conv_bias, conv_ops, add_z, use_tracing):
+        with set_default_dtype(torch.float):
```

Comment:
For reviewers: The diff formatting is terrible here.  I added a context manager to change the default dtype (see _many_ similar examples elsewhere in this file), and indented the rest of the test.  I did the same in the next test.  As a result, this was done in two consecutive tests, and the diff was rendered in a way that obfuscates the change.

The changes to focus on:
- Calling new miopen routine if TEST_WITH_ROCM.
- Tests are indented inside context manager to temporarily set the default dtype.  MIOpen does not support double type convolutions.

Reply:
Viewing with "hide whitespace" renders with only the relevant changes.